Gems/Profiler/Code/Source/CpuProfiler.cpp

```diff
@@ -98,9 +98,10 @@ namespace Profiler
             {
                 // Lazy initialization, creates an instance of the Thread local data if it's not created, and registers it
                 RegisterThreadStorage();
-
+                va_list args;
+                va_start(args, eventNameArgCount);
                 // Push it to the stack
-                CachedTimeRegion timeRegion({budget->Name(), eventName});
+                CachedTimeRegion timeRegion({ budget->Name(), AZStd::string::format_arg(eventName, args).c_str() });
```

Comment:
This would be better to use a `AZStd::fixed_string<512>::format_arg` to avoid any heap allocations